### PR TITLE
Improvements to tracing solution

### PR DESCRIPTION
**Pull request by ceelias**

### CODE REVIEW

File: monitoring/bin/deploy_monitoring_openshift.sh [hunk 1]
```diff
@@ -261,6 +274,33 @@ if ! kubectl get route -n $MON_NS v4m-grafana 1>/dev/null 2>&1; then
...
+  log_info "Tracing enabled..."
+  log_info "Installing tempo"
+  helm upgrade --install v4m-tempo \
+    -n "$MON_NS" \
+    -f "$TEMPO_USER_YAML" \
+    -f "$airgapValuesFile" \
+    --set serviceMonitor.enabled=true \
+    --set searchEnabled=true \
+    --version "$TEMPO_CHART_VERSION" \
+    ${AIRGAP_HELM_REPO}grafana/tempo
```
Comment (ceelias) [inline monitoring/bin/deploy_monitoring_openshift.sh:301]: Will update this to use `chart2install`

File: monitoring/airgap/airgap-tempo-values.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+image:
```
Comment (gsmith-sas) [inline monitoring/airgap/airgap-tempo-values.yaml:1]: Looking at the [Temp Helm chart, ](https://github.com/grafana/helm-charts/blob/6e2172f08b5539384734a42262f42836b8075fc5/charts/tempo/values.yaml#L16) it looks like the correct key  for this is NOT **image.repository** but **tempo.repository**.  Or, am I mis-reading things?

File: monitoring/airgap/airgap-tempo-values.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  repository: __AIRGAP_REGISTRY__/tempo/tempo
+  pullSecrets:
+  - name: __AIRGAP_IMAGE_PULL_SECRET_NAME__
```
Comment (gsmith-sas) [inline monitoring/airgap/airgap-tempo-values.yaml:4]: See above; the correct key may be **tempo.pullSecrets** (although I also see a **serviceAccount.imagePullSecret** key).  Please confirm we're setting the correct keys.

File: monitoring/bin/deploy_monitoring_cluster.sh [hunk 4]
```diff
@@ -258,6 +258,20 @@ log_info "Deploying ServiceMonitors and Prometheus rules"
...
+    # Check for the image pull secret for the air gap environment and replace placeholders
+    checkForAirgapSecretInNamespace "$AIRGAP_IMAGE_PULL_SECRET_NAME" "$MON_NS"
+    replaceAirgapValuesInFiles "monitoring/airgap/airgap-tempo-values.yaml"
+
+    airgapValuesFile=$updatedAirgapValuesFile
+  else
+    airgapValuesFile=$TMP_DIR/empty.yaml
+  fi
+
+  TEMPO_CHART_VERSION=${TEMPO_CHART_VERSION:-1.5.0}
```
Comment (gsmith-sas) [inline monitoring/bin/deploy_monitoring_cluster.sh:274]: Take a look at the changes I just merged in as part of #540.  They include some changes to this file.

File: monitoring/bin/deploy_monitoring_openshift.sh [hunk 5]
```diff
@@ -261,6 +274,33 @@ if ! kubectl get route -n $MON_NS v4m-grafana 1>/dev/null 2>&1; then
...
+    # Check for the image pull secret for the air gap environment and replace placeholders
+    checkForAirgapSecretInNamespace "$AIRGAP_IMAGE_PULL_SECRET_NAME" "$MON_NS"
+    replaceAirgapValuesInFiles "monitoring/airgap/airgap-tempo-values.yaml"
+
+    airgapValuesFile=$updatedAirgapValuesFile
+  else
+    airgapValuesFile=$TMP_DIR/empty.yaml
+  fi
+
+  TEMPO_CHART_VERSION=${TEMPO_CHART_VERSION:-1.5.0}
```
Comment (gsmith-sas) [inline monitoring/bin/deploy_monitoring_openshift.sh:291]: See note above re:changes merged in as part #540 .

**Review by gsmith-sas — CHANGES_REQUESTED** [review]
We need to double-check some Helm chart keys and reconcile these changes with those pushed in #540.

**Review by gsmith-sas — CHANGES_REQUESTED** [review]
Looks good but needs some tweaks, especially wrt serviceAccounts and automounting of tokens.

File: monitoring/values-tempo.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+serviceMonitor:
+  enabled: true
+  interval: ""
+  additionalLabels: {}
+  annotations: {}
+  # scrapeTimeout: 10s
+
+persistence:
+  enabled: false
```
[the remark this answers is not in the document]
Comment (ceelias) [reply monitoring/values-tempo.yaml:67]: From what I can tell this is sep from the 24 hour persistence, this one will create a PVC if enabled.

File: monitoring/openshift/tempo-values.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        http:
+          endpoint: "0.0.0.0:4318"
+
+serviceAccount:
+  # -- Specifies whether a ServiceAccount should be created
+  create: true
+  # -- The name of the ServiceAccount to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: null
+  automountServiceAccountToken: true
```
[the remark this answers is not in the document]
Comment (ceelias) [reply monitoring/openshift/tempo-values.yaml:52]: Interestingly, I set this to false and I still see that the pods are mounting the service account. Am I missing something?

**Review by gsmith-sas — APPROVED** [review]
LGTM

**Pull request merged**